pkg/daemon/ceph/osd/volume.go

```diff
@@ -211,11 +212,19 @@ func getCephVolumeOSDs(context *clusterd.Context, clusterName string) ([]oposd.O
 		var osdFSID string
 		isFilestore := false
 		for _, osd := range osdInfo {
+			if osd.Tags.ClusterFSID != cephfsid {
+				logger.Infof("skipping osd%d: %s running on a different ceph cluster: %s", id, osd.Tags.OSDFSID, osd.Tags.ClusterFSID)
+				continue
+			}
 			osdFSID = osd.Tags.OSDFSID
 			if osd.Type == "journal" {
 				isFilestore = true
 			}
 		}
+		if len(osdFSID) == 0 {
```

Comment:
Added this case to fix the following scenario:
"Two ceph clusters -> On Single Node k8 cluster -> Ceph cluster 1 has osd on sda and sdb > ceph cluster 2 has osds on sdc"